include/spdlog/sinks/ringbuffer_sink-inl.h

```diff
@@ -0,0 +1,46 @@
+// Copyright(c) 2015-present, Gabi Melman & spdlog contributors.
+// Distributed under the MIT License (http://opensource.org/licenses/MIT)
+
+#pragma once
+
+#ifndef SPDLOG_HEADER_ONLY
+#include "spdlog/sinks/ringbuffer_sink.h"
+#endif
+
+#include "spdlog/common.h"
+#include "spdlog/details/os.h"
+
+namespace spdlog {
+namespace sinks {
+
+template<typename Mutex>
+SPDLOG_INLINE ringbuffer_sink<Mutex>::ringbuffer_sink(size_t buf_size)
+{
+    buf=details::circular_q<std::string>(buf_size);
```

Comment:
Using circular_q<log_msg_buffer> instead of atring will improve performance:
1: Avoids allocation of string in each message
2. format() can be avoided until really needed (in the last(size_t lim) method)
 see https://github.com/gabime/spdlog/blob/v1.x/include/spdlog/details/backtracer.h as an example

Reply:
I tried first with `log_msg` but deferred formatting was garbage as presumably some of the data were out of scope already. That's why I went the `std::string` way with perf impact. If I use `log_msg_buffer`, this issue will not happen? If so, good to hear and I will be happy to change that.

Reply:
> If I use log_msg_buffer, this issue will not happen?

No, it won't happen. That is the sole purpose of ```log_msg_buffer```. To deep copy log_messages off the stack into it's internal buffer.